gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleProject.java

```diff
@@ -242,4 +243,23 @@ private GradleRunConfigurationAttributes getRunConfigurationAttributes( String t
             showExecutionView, showConsoleView );
     }
 
+    @Override
+    public <T> T adapt( Class<T> adapterType )
```

Comment:
This wizard should only be generating new java files.  It should never mess
with dependencies.  We can not make this stable without writing a
build.gradle parser whic we are not going to do.  So there should be no
code in gradle.core at all.

On Tue, Mar 8, 2016 at 6:55 PM, simon.jiang <notifications@github.com>
wrote:

> In
> gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/LiferayGradleProject.java
> <https://github.com/gamerson/liferay-ide/pull/1433#discussion_r55456680>:
>
> > @@ -242,4 +243,23 @@ private GradleRunConfigurationAttributes getRunConfigurationAttributes( String t
> >              showExecutionView, showConsoleView );
> >      }
> >
> > +    @Override
> > +    public <T> T adapt( Class<T> adapterType )
>
> We have two reason: 1)Because after we changed build.gradle, we need to
> call SynchronizeGradleProjectsJob to refresh gradle proejct for all types
> new ds component , otherwise gradle project didn't recognize the new
> dependencies. User will have to refresh manually. We have to put the
> implements of AbstractLiferayModuleOperation in gradle core plugin. The
> project core can't invoke gradle core function. What do you means the
> DSCOmponents completely inside project core? 2) Do you mean we won't
> support to add new maven ds component in future or maven and gradle will
> use same solution to add new ds component?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gamerson/liferay-ide/pull/1433/files#r55456680>.
>



-- 
Greg Amerson
Liferay Developer Tools
Liferay, Inc. www.liferay.com